### PR TITLE
[DERCBOT-768] AI Orchestrator API - Check LLM/EM provider setting

### PR DESCRIPTION
**Pull request by assouktim**

### CODE REVIEW

**Comment by assouktim** [comment]
I think we should rename the **_LLM Orchestrator_** project to **_AI Orchestrator_**.

**Review by Benvii — CHANGES_REQUESTED** [review]
Check settings, invocation part and vector generation of a test string can be in the abstraction layer as they will always be the same

File: llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/errors/error_factories.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#
+from fastapi import Request
+from openai import APIError
+
+from llm_orchestrator.errors.exceptions.exceptions import BusinessException
+from llm_orchestrator.models.errors.errors_models import ErrorInfo
+from llm_orchestrator.routers.responses.responses import ErrorResponse
+
+
+def create_error_info_openai(exc: APIError, provider: str) -> ErrorInfo:
```
[the remark this answers is not in the document]
Comment (Benvii) [reply llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/errors/error_factories.py:23]: Ok I just saw that APIError is from openai, maybe it should be located in an openai specific file ?

File: llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/models/errors/errors_models.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+from enum import Enum, unique
+from typing import Optional
+
+from pydantic import BaseModel, Field
+
+from llm_orchestrator.models.llm.llm_provider import LLMProvider
+
+
+@unique
+class ErrorCode(Enum):
```
[the remark this answers is not in the document]
Comment (Benvii) [reply llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/models/errors/errors_models.py:26]: Committed

**Review by Benvii — APPROVED** [review]
Thanks for this good rand rich PR, it's good for me, don't hesitate to check my small commit about the enum documentation.

**Pull request merged**